core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java

```diff
@@ -253,38 +260,63 @@ public void removePath() throws Exception {
 
   @Test
   public void metricCacheSize() throws Exception {
-    Gauge<Long> cacheSize = (Gauge<Long>) MetricsSystem.METRIC_REGISTRY.getGauges()
-        .get(MetricKey.MASTER_ABSENT_CACHE_SIZE.getName());
-    mUfsAbsentPathCache.addSinglePath(new AlluxioURI("/mnt/1"));
+    MetricsSystem.resetCountersAndGauges();
+    ManualClock clock =
+        new ManualClock(AsyncUfsAbsentPathCache.METRIC_CACHE_TIMEOUT_SEC + 1, TimeUnit.SECONDS);
+    PowerMockito.mockStatic(Clock.class);
```

Comment:
I'm thinking this mock may be fragile because it relies on how the gauge looks at time. Since you have the static variable already, can you set this to a super small number in the test and sleep between peeking at the cached gauge, to make sure the cache goes away?

Reply:
You mean make `AsyncUfsAbsentPathCache.METRIC_CACHE_TIMEOUT_SEC` mutable? And set it in UTs?

Reply:
oops it's impossible for Powermock or reflection to update a static final constant at runtime. So you can do this:

```
int getCachedGaugeTimeoutSec() {
  return 2;
}
```
then
```
class TestAsyncUfsAbsentPathCache extends AsyncUfsAbsentPathCache {
  @Override
  int getCachedGaugeTimeoutSec() {
    // not sure what happens if you return 0 here
    // but if caching 0 sec is invalid argument, you can make the method based on millisecond and return a 1ms
  }
}
```

Reply:
Sounds good.